packages/hoppscotch-app/src/helpers/utils/EffectiveURL.ts

```diff
@@ -280,7 +279,9 @@ function getFinalBodyFromRequest(
       ),
       toFormData
     )
-  } else return parseBodyEnvVariables(request.body.body, envVariables)
+  }
+
+  return parseTemplateString(request.body.body, envVariables)
```

Comment:
We have `parseBodyEnvVariables` which handle a special case we have with body  where if we don't get a match, we will leave the environment variables as is and don't parse them down like we do in other places.

Hence, what is to be fixed is not here, but on the `parseBodyEnvVariables` function.